test/e2e/app-dir/css-order/css-order.test.ts

```diff
@@ -205,6 +204,118 @@ const PAGES: Record<
   },
 }
 
+const SIDE_EFFECTS_PAGES: Record<
+  string,
+  {
+    url: string
+    selector: string
+    color: string
+    background?: string
+    skip?: Array<'turbo' | 'loose' | 'strict'>
+  }
+> = {
+  'vendor-side-effects-array': {
+    url: '/vendor/a',
+    selector: '#vendor-side-effects-array',
+    background: 'rgb(0, 254, 0)',
+    color: 'rgb(254, 0, 0)',
+    /**
+     * TURBOPACK: This should be supported by turbo but does not appear to be today,
+     * because the emitted rules are not in the correct order when this is set.
+     * Results are inconsistent so we cannot reliably test against without
+     * removing this skip...
+     */
+    skip: ['turbo'],
+  },
+  'vendor-side-effects-true': {
+    url: '/vendor/b',
+    selector: '#vendor-side-effects-true',
+    background: 'rgb(0, 253, 0)',
+    color: 'rgb(253, 0, 0)',
+  },
+  'vendor-side-effects-false': {
+    url: '/vendor/c',
+    selector: '#vendor-side-effects-false',
+    background: 'rgb(0, 252, 0)',
+    color: 'rgb(252, 0, 0)',
+    /**
+     * Turbopack has buttoned up a webpack bug.
+     * Packages *should* side-effect css modules, but webpack doesn't retain
+     * the proper import order context of css modules in libraries within
+     * node_modules when side-effects is set to true.
+     * Related webpack bug: https://github.com/webpack/webpack/issues/7094
+     */
+    skip: ['turbo'],
+  },
+  'vendor-side-effects-global-array': {
+    url: '/vendor/d',
+    selector: '#vendor-side-effects-global-array',
+    background: 'rgb(0, 250, 0)',
+    color: 'rgb(250, 0, 0)',
+    /**
+     * TURBOPACK: This should be supported by turbo but does not appear to be today,
+     * because the emitted rules are not in the correct order when this is set.
+     * Results are inconsistent so we cannot reliably test against without
+     * removing this skip...
+     */
+    skip: ['turbo'],
+  },
+  'pages-vendor-side-effects-array': {
+    url: 'pages/vendor/a',
+    selector: '#vendor-side-effects-array',
+    background: 'rgb(0, 254, 0)',
+    color: 'rgb(254, 0, 0)',
+    /**
+     * TURBOPACK: This should be supported by turbo but does not appear to be today,
+     * because the emitted rules are not in the correct order when this is set.
+     * Results are inconsistent so we cannot reliably test against without
+     * removing this skip...
+     *
+     * WEBPACK: css modules are inconsistent when included as sideEffects
+     * when not Boolean `true` or `false`
```

Comment:
Weird that it works with `"sideEffects": false` in package.json. That should have the same effect.

But technically it's within the spec'ed behavior. Flagged as module as side effect free also means it could be included in any order (since it has no side effects). That's why the idea was to force side effects to be true for any CSS related thing since they actually have the import order as side effect.

Reply:
What I believe to be happening is with webpack when it's false there is still _some_ instability, but it is following each first tree reference deeply, seeing those stylesheets, adding them, and climbing back to the initial file and adding those stylesheets.

When only stylesheets are being included in side effects, the JS tree is being ignored and the CSS files are being added in order that they are run into (top of tree to bottom, a complete inversion and broken behavior). When true, the CSS files are added according to the JS tree from bottom to top in full library order, rather than semi-random order like false.